File: backend/src/main/java/com/github/kevinldg/mygameslist/backend/game/Game.java
package com.github.kevinldg.mygameslist.backend.game;

public record Game(
        int gameId,
        String gameName,
        String gameSummary,
        int artworkId,
        String artworkUrl,
        GameState gameState
) {
    public Game withGameState(GameState gameState) {
        return new Game(this.gameId(), this.gameName(), this.gameSummary(), this.artworkId(), this.artworkUrl, gameState);
    }
}

File: frontend/src/types/Game.ts
import {GameState} from "../enums/GameState.ts";

export type Game = {
    gameId: number;
    gameName: string;
    gameSummary: string;
    artworkId: number;
    artworkUrl: string;
    gameState: GameState;
};

export type GameEntryProps = {
    game: Game;
    updateGame?: (gameName: string, gameState: GameState) => void;
    deleteGame?: (gameName: string) => void;
    favorGame?: (game: Game) => void;
    dontShowGameState?: boolean;
}

export type GameStateBadgeProps = {
    updateGame?: (gameName: string, gameState: GameState) => void;
    game?: Game;
    gameState: GameState;
}